haskell/src/Monocle/Backend/Queries.hs

```diff
@@ -312,23 +317,27 @@ getTermsAgg query onTerm = do
     isNotEmptyTerm :: BH.TermsResult -> Bool
     isNotEmptyTerm tr = getTermKey tr /= ""
 
-data TermResult = TermResult {term :: Text, count :: Int} deriving (Show, Eq)
-
-getRepos :: QueryM [TermResult]
-getRepos = do
+getDocTypeTopCountByField :: Text -> Text -> QueryFlavor -> QueryM [TermResult]
+getDocTypeTopCountByField doctype attr qflavor = do
   -- Prepare the query
-  basequery <- toBHQueryWithFlavor (QueryFlavor OnAuthor CreatedAt) <$> getQuery
-  let query = mkAnd $ basequery <> getQueryE
+  basequery <- toBHQueryWithFlavor qflavor <$> getQuery
+  let query = mkAnd $ basequery <> [documentType doctype]
```

Comment:
nit: this pattern is so common, there is a generalized solution with [withFilter](https://github.com/change-metrics/monocle/blob/aefaff60f5c162f3d603a3f54d569ca77bcd7cce/haskell/src/Monocle/Servant/Env.hs#L122) and [getQueryBHWithFlavor](https://github.com/change-metrics/monocle/blob/aefaff60f5c162f3d603a3f54d569ca77bcd7cce/haskell/src/Monocle/Servant/Env.hs#L109). They can be used like so here:

```haskell
getDocTypeTopCountByField :: Text -> Text -> QueryFlavor -> QueryM [TermResult]
getDocTypeTopCountByField doctype attr qflavor = withFilter [documentType doctype] $ do
    query <- getQueryBHWithFlavor qflavor

    results <- liftTenantM (runTermAgg query)
    ...
```